x-pack/plugins/security/public/views/login/components/basic_login_form.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiCallOut, EuiFieldText, EuiFormRow, EuiPanel, EuiSpacer } from '@elastic/eui';
+import React, { ChangeEvent, Component, Fragment } from 'react';
+import { LoginState } from '../../../../common/login_state';
+
+interface Props {
+  http: any;
+  window: any;
+  infoMessage?: string;
+  loginState: LoginState;
+  next: string;
+}
+
+interface State {
+  hasError: boolean;
+  isLoading: boolean;
+  username: string;
+  password: string;
+  message: string;
+}
+
+export class BasicLoginForm extends Component<Props, State> {
+  public state = {
+    hasError: false,
+    isLoading: false,
+    username: '',
+    password: '',
+    message: '',
+  };
+
+  public componentDidMount() {
+    document.addEventListener('keydown', this.submitOnEnter);
```

Comment:
Just wrap the form in an actual `<form>` element, make the button below `type="submit"` and make the `this.submit` method be called via `<form onSubmit={this.submit}>` instead. No need for making manual key listeners here.

Reply:
Great call, that's much simpler, and the way forms have always worked 👍